[Hub Generated] Review request for Microsoft.OffAzure to add version stable/2023-06-06

tarsharm opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/tarsharm-Microsoft.OffAzure-2023-06-06...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/b321e1dc-5e07-4940-9cd6-a65ac6729a8b)

[1] [public repo review queue], [private repo review queue]  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue]  

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[public repo review queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo review queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

tarsharm committed:
Adds base for updating Microsoft.OffAzure from version stable/2020-07-07 to version 2023-06-06

tarsharm committed:
Updates readme

tarsharm committed:
Updates API version in new specs and examples

tarsharm committed:
Initial swagger check-in

tarsharm committed:
Fix Avocado errors

tarsharm committed:
Fix breaking changes in swagger

tarsharm committed:
resubmitting json again

tarsharm committed:
Resolving breaking change across version exception

tarsharm committed:
Revert casing changes

tarsharm committed:
Revert "Revert casing changes"

This reverts commit ab8d2082cf3e5817e8bb9f7e88e228ad629d131a.

tarsharm committed:
Revert "Resolving breaking change across version exception"

This reverts commit 3b649380472af20fbdaf99369642058e7b1f9ec9.

tarsharm committed:
Revert "resubmitting json again"

This reverts commit 3a1e982bf4e0e6860a8775676b70a1bf6f704445.

tarsharm committed:
Revert "Fix breaking changes in swagger"

This reverts commit b586b353aa0b06a607d2a707c190458ec6ac2c77.

markcowl reviewed (changes requested):
Before looking at this very large PR, a few issues:

- it looks like there are multiple armProvider namespaces, which indicate different provider names which should go in completely different directories. Only some of these appear to have generated swagger.
- and multiple tspconfig.yaml files - the standard is to have a single tspConfig.yaml and a single main.tsp as entry point.  We need to understand your intention, but it looks like many of these are unused in generating swagger.

- you should not check in a common-types directory for your swagger
- it looks like you are checking in a massive number of api-versions at once - why is this the case?

TimLovellSmith commented:
@tarsharm This is a required step:

>I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
[ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
[REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).
I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

tarsharm commented:
> * it looks like there are multiple armProvider namespaces, which indicate different provider names which should go in completely different directories. Only some of these appear to have generated swagger.
> *

@markcowl , 
1. Correct there are multiple namespaces Migrate and OffAzure. Swaggers for other namespace Migrate will be added in another PR.
2. Removed the other tspconfig.yaml file.
3. Removed Common types folder.
4. I added you in th email chain. Swagger is not updated for last 3 years but multiple versions were GA'ed. We now want to publish swagger for just one version 2023-06-06 and will depricate for all others.

tarsharm committed:
Take review comments and remove casing breaking changes

tarsharm committed:
Fix LintDiff Exception
